packages/fuel-gauge/src/transaction-response.test.ts

```diff
@@ -9,6 +9,75 @@ import {
   randomBytes,
   WalletUnlocked,
 } from 'fuels';
+import type { MockInstance } from 'vitest';
+
+async function verifyKeepAliveMessageWasSent(subscriptionStream: ReadableStream<Uint8Array>) {
+  const decoder = new TextDecoder();
+  const reader = subscriptionStream.getReader();
+  let hasKeepAliveMessage = false;
+  do {
+    const { value, done } = await reader.read();
+    if (done) {
+      break;
+    }
+    const text = decoder.decode(value);
+    if (text === ':keep-alive-text\n\n') {
+      hasKeepAliveMessage = true;
+    }
+  } while (!hasKeepAliveMessage);
+
+  // The keep-alive message is sent every 15 seconds,
+  // and this assertion verifies that it was indeed sent.
+  // if this fails, check if the duration was changed on the fuel-core side.
+  // As of the time of writing, the latest permalink where this info can be found is:
+  // https://github.com/FuelLabs/fuel-core/blob/bf1b22f47c58a9d078676c5756c942d839f38916/crates/fuel-core/src/graphql_api/service.rs#L247
+  // To get the actual latest info you need to check out the master branch:
+  // https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-core/src/graphql_api/service.rs#L247
+  // This link can fail because master can change.
+  expect(hasKeepAliveMessage).toBe(true);
+}
+
+function getSubscriptionStreamFromFetch(streamHolder: { stream: ReadableStream<Uint8Array> }) {
+  function getFetchMock(
+    fetchSpy: MockInstance<
+      [input: RequestInfo | URL, init?: RequestInit | undefined],
+      Promise<Response>
+    >
+  ) {
+    return async (...args: Parameters<typeof fetch>) => {
+      /**
+       * We need to restore the original fetch implementation so that fetching is possible
+       * We then get the response and mock the fetch implementation again
+       * So that the mock can be used for the next fetch call
+       */
+      fetchSpy.mockRestore();
+      const r = await fetch(...args);
+      fetchSpy.mockImplementation(getFetchMock(fetchSpy));
+
+      const isSubscriptionCall = args[0].toString().endsWith('graphql-sub');
+      if (!isSubscriptionCall) {
+        return r;
+      }
+
+      /**
+       * Tee duplicates a stream and all writes happen to both streams.
```

Comment:
```suggestion
       * This duplicates a stream and all writes happen to both streams.
```

Reply:
This is correct, it's explaining what `r.body!.tee();` below does.